system/MenuButton.tsx

```diff
@@ -0,0 +1,37 @@
+import styles from '@system/MenuButton.module.scss';
+
+import * as React from 'react';
+
+import { useModal } from '@system/providers/ModalContextProvider';
+
+export default function MenuButton(props) {
+  const { showModal } = useModal();
+
+  const [menuActive, setMenuActive] = React.useState(false);
+
+  const hamburgerRef = React.useRef<HTMLDivElement>(null);
+
+  // (@xBalbinus): The hamburger menu gets closed on click to any outside HTML element if we only use
+  // onOutsideEvent, we need to check if the click is coming from the hamburger button specifically.
+  function toggleModal() {
+    if (hamburgerRef.current) {
+      hamburgerRef.current.classList.toggle(styles.active);
+    }
```

Comment:
this class will not survive re-renders. if the menu is re-rendered for any reason and active is set, the class won't be there anymore.

should set `className` based on `menuActive` instead. e.g. `className={Utilities.classNames(styles.hamburger, menuActive && styles.active)}`